src/targets/gpu/nonzero.cpp

```diff
@@ -8,6 +8,7 @@ namespace gpu {
 
 shape hip_nonzero::compute_shape(std::vector<shape> inputs) const
 {
+    check_shapes({inputs.front()}, *this).has(1).standard();
```

Comment:
The device function will have problem if input is a broadcasted shape for the nonzero gpu implementation.